include/api.php

```diff
@@ -3276,72 +3276,706 @@ function api_oauth_access_token($type) {
-				FROM `photo`
-				WHERE `uid` = %d AND `album` != 'Contact Photos' GROUP BY `resource-id`, `album`, `filename`, `type`",
+		// now let's delete all photos from the album
+		$result = q("DELETE FROM `photo` WHERE `uid` = %d AND `album` = '%s'",
+				intval(api_user()),
+				dbesc($album));
+
+		// return success of deletion or error message
+		if ($result) {
+			$answer = array('result' => 'deleted', 'message' => 'album `' . $album . '` with all containing photos has been deleted.');
+			return api_format_data("photoalbum_delete", $type, array('$result' => $answer));
+		}
+		else {
+			throw new BadRequestException("unknown error");
+		}
+
+	}
+
+	/**
+	 * @brief update the name of the album for all photos of an album
+	 *
+	 * @param string $type Known types are 'atom', 'rss', 'xml' and 'json'
+	 * @return string
+	 */
+	function api_fr_photoalbum_update($type) {
+		if (api_user()===false)
+			throw new ForbiddenException();
+
+		// input params
+		$album = (x($_REQUEST,'album') ? $_REQUEST['album'] : "");
+		$album_new = (x($_REQUEST,'album_new') ? $_REQUEST['album_new'] : "");
+
+		// we do not allow calls without album string
+		if ($album == "")
+			throw new BadRequestException("no albumname specified");
+		if ($album_new == "")
+			throw new BadRequestException("no new albumname specified");
+
+		// check if album is existing
+		$r = q("SELECT * FROM `photo` WHERE `uid` = %d AND `album` = '%s'",
+				intval(api_user()),
+				dbesc($album));
+		if (!dbm::is_result($r))
+			throw new BadRequestException("album not available");
+
+		// now let's update all photos to the albumname
+		$result = q("UPDATE `photo` SET `album` = '%s' WHERE `uid` = %d AND `album` = '%s'",
+				dbesc($album_new),
+				intval(api_user()),
+				dbesc($album));
+
+		// return success of updating or error message
+		if ($result) {
+			$answer = array('result' => 'updated', 'message' => 'album `' . $album . '` with all containing photos has been renamed to `' . $album_new . '`.');
+			return api_format_data("photoalbum_update", $type, array('$result' => $answer));
+		}
+		else {
+			throw new BadRequestException("unknown error");
+		}
+	}
+
+
+	/**
+	 * @brief list all photos of the authenticated user
+	 *
+	 * @param string $type Known types are 'atom', 'rss', 'xml' and 'json'
+	 * @return string
+	 */
+	function api_fr_photos_list($type) {
+		if (api_user()===false) throw new ForbiddenException();
+		$r = q("select `resource-id`, max(scale) as scale, album, filename, type, max(created) as created,
+				max(edited) as edited, max(`desc`) as `desc` from photo
+				where uid = %d and album != 'Contact Photos' group by `resource-id`",
 			intval(local_user())
 		);
 		$typetoext = array(
-			'image/jpeg' => 'jpg',
-			'image/png' => 'png',
-			'image/gif' => 'gif'
+		'image/jpeg' => 'jpg',
+		'image/png' => 'png',
+		'image/gif' => 'gif'
 		);
-		$data = array('photo' => array());
-		if (dbm::is_result($r)) {
+		$data = array('photo'=>array());
+		if ($r) {
```

Comment:
was no intention, when I started with the code, it was still "if ($r)", I just copied it over. changed to dbm::is...